### PR TITLE
Ignore root nodes when exporting to GLB

### PR DESCRIPTION
**Pull request by noalak**
Ignore the root nodes used for conversion from right-handed to left-handed when exporting to GLB.

Solving issue #6349.

It implements Gary's solution discussed in last PR https://github.com/BabylonJS/Babylon.js/pull/7338.

PR is split in 2 commits:
1) First commit for ignoring **one** root node
2) Second commit to ignore **multiple** root nodes

It might be easier to look at file changed in 1st commit first.

Regarding implementation, the __convertToRightHandedSystemMap_ is initialized with all nodes. You may prefer to have a default value used when Map.get returns undefined.
Right now, it's sometime necessary to cast to boolean in case the node is not found (should never happen): `convertToRightHandedSystemMap.get(babylonTransformNode) as boolean`

@bghgary Thanks for the hints! Could you review the PR ?

### CODE REVIEW

**Comment by noalak** [comment]
Looking to fix Unit tests (lights)

**Comment by noalak** [comment]
Added a check on node type. The following node types are not candidate for a root node used to convert from right-handed to left-handed:
- lights
- cameras
- mesh, or instance of a mesh, with geometry

File: serializers/src/glTF/2.0/glTFExporter.ts [hunk 1]
```diff
@@ -1293,6 +1302,23 @@ export class _Exporter {
...
+     * Check if the node is used to convert its descendants from right-handed to left-handed
+     * @param node The node to check
+     * @returns True if the node is used to convert its descendants from right-handed to left-handed. False otherwise
+     */
+    private isNodeConvertingToLeftHanded(node: Node): boolean {
+        if (node instanceof TransformNode &&
+            (!node.position.equalsToFloats(0, 0, 0) ||
+            (!node.rotationQuaternion && node.rotation && (node.rotation.x != 0 || node.rotation.z != 0 || Math.abs(node.rotation.y - Math.PI) > Epsilon)) || // rotation Quaternion has priority over Vector3
+            (node.rotationQuaternion && !node.rotationQuaternion.equals(new Quaternion(0, 1, 0, 0))) ||
+            !node.scaling.equalsToFloats(1, 1, -1))) {
```
[the remark this answers is not in the document]
Comment (noalak) [reply serializers/src/glTF/2.0/glTFExporter.ts]: Done!

File: serializers/src/glTF/2.0/glTFExporter.ts [hunk 2]
```diff
@@ -122,9 +122,9 @@ export class _Exporter {
     private _nodeMap: { [key: number]: number };
 
     /**
-     * Specifies if the Babylon scene should be converted to right-handed on export
+     * Specifies if a Babylon node should be converted to right-handed on export
      */
-    public _convertToRightHandedSystem: boolean;
+    public _convertToRightHandedSystemMap: Map<Node, boolean>;
```
[the remark this answers is not in the document]
Comment (noalak) [reply serializers/src/glTF/2.0/glTFExporter.ts]: Done!

**Comment by deltakosh** [comment]
Last check @bghgary ?

File: serializers/src/glTF/2.0/glTFExporter.ts [hunk 3]
```diff
@@ -1293,6 +1302,36 @@ export class _Exporter {
...
+                if (Math.abs(matrixProduct.m[i] - matrixIdentity.m[i]) > Epsilon) {
+                    return false;
+                }
+            }
+
+            // Geometry
+            if ((node instanceof Mesh && node.geometry !== null) ||
+                (node instanceof InstancedMesh && node._masterMesh instanceof Mesh && node._masterMesh.geometry !== null)) {
+                return false;
+            }
```
Comment (bghgary) [inline serializers/src/glTF/2.0/glTFExporter.ts]: Do you still need this check?

Comment (noalak) [reply serializers/src/glTF/2.0/glTFExporter.ts]: Well do we want to exclude a root cube from being exported if it somehow has this specific transform ?
I think we should only exclude dummies. The glTF loader is creating a BABYLON.Mesh.

File: serializers/src/glTF/2.0/glTFExporter.ts [hunk 4]
```diff
@@ -1293,6 +1302,36 @@ export class _Exporter {
...
+     * @returns True if the node is used to convert its descendants from right-handed to left-handed. False otherwise
+     */
+    private isNodeConvertingToLeftHanded(node: Node): boolean {
+        if (node instanceof TransformNode)
+        {
+            // Transform
+            let matrix = node.getWorldMatrix();
+            let matrixToLeftHanded = Matrix.Compose(new Vector3(-1, 1, 1), Quaternion.Identity(), Vector3.Zero());
+            let matrixProduct = matrix.multiply(matrixToLeftHanded);
+            let matrixIdentity = Matrix.Identity();
```
[the remark this answers is not in the document]
Comment (noalak) [reply serializers/src/glTF/2.0/glTFExporter.ts]: Ok, I will update the PR tomorrow

File: serializers/src/glTF/2.0/glTFExporter.ts (hunk 3 above)
Comment (bghgary) [reply serializers/src/glTF/2.0/glTFExporter.ts]: Oh, I see. This is checking to make sure there is no geometry.

File: serializers/src/glTF/2.0/glTFExporter.ts [hunk 5]
```diff
@@ -1293,6 +1302,36 @@ export class _Exporter {
...
+            for (let i = 0; i < 16; i++) {
+                if (Math.abs(matrixProduct.m[i] - matrixIdentity.m[i]) > Epsilon) {
+                    return false;
+                }
+            }
+
+            // Geometry
+            if ((node instanceof Mesh && node.geometry !== null) ||
+                (node instanceof InstancedMesh && node._masterMesh instanceof Mesh && node._masterMesh.geometry !== null)) {
```
Comment (bghgary) [inline serializers/src/glTF/2.0/glTFExporter.ts]: Probably mean `sourceMesh` here.

```suggestion
                (node instanceof InstancedMesh && node.sourceMesh.geometry !== null)) {
```

File: serializers/src/glTF/2.0/glTFExporter.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (noalak) [reply serializers/src/glTF/2.0/glTFExporter.ts]: Done

File: serializers/src/glTF/2.0/glTFExporter.ts (hunk 5 above)
Comment (noalak) [reply serializers/src/glTF/2.0/glTFExporter.ts]: Done

**Comment by noalak** [comment]
All updates done.

@bghgary Your contribution was greatly appreciated!

**Pull request merged**